op-node/rollup/driver/state.go

```diff
@@ -278,11 +280,21 @@ func (s *Driver) eventLoop() {
 				return
 			}
 			if s.network != nil && payload != nil {
-				// Publishing of unsafe data via p2p is optional.
-				// Errors are not severe enough to change/halt sequencing but should be logged and metered.
-				if err := s.network.PublishL2Payload(s.driverCtx, payload); err != nil {
-					s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
-					s.metrics.RecordPublishingError()
+				isActive, err := s.sequencerConductor.Active(s.driverCtx)
+				if err != nil {
+					s.log.Error("failed to determine if this node is the active sequencer, skipping publishing of unsafe data", "err", err)
+				} else if isActive {
+					// Publishing of unsafe data via p2p is optional.
+					// Errors are not severe enough to change/halt sequencing but should be logged and metered.
+					if err := s.network.PublishL2Payload(s.driverCtx, payload); err != nil {
+						s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
+						s.metrics.RecordPublishingError()
+					}
+					if err := s.sequencerConductor.CommitUnsafePayload(s.driverCtx, payload); err != nil {
+						s.log.Error("failed to commit unsafe payload to conductor log", "id", payload.ID(), "err", err)
+					}
+				} else {
+					s.log.Warn("Not the active sequencer, skipping publishing of unsafe data")
```

Comment:
I don't believe this is needed since it's called within `<-sequencerCh` which implies that only active sequencer will enter here.

Reply:
i think if we are checking leader sequencer (not active) this is still useful.